core/drivers/crypto/se050/adaptors/apis/apdu.c

```diff
@@ -0,0 +1,525 @@
+	if (status != SM_OK)
+		return kStatus_SSS_Fail;
+
+	status = Se05x_API_WriteRSAKey(s_ctx, NULL, k_object->keyId,
+				       0,
+				       SE05X_RSA_NO_p,
+				       SE05X_RSA_NO_q,
+				       SE05X_RSA_NO_dp,
+				       SE05X_RSA_NO_dq,
+				       SE05X_RSA_NO_qInv,
+				       SE05X_RSA_NO_pubExp,
+				       SE05X_RSA_NO_priv,
+				       keypair->n, keypair->n_len,
+				       (SE05x_INS_t)type,
+				       kSE05x_KeyPart_NA,
+				       kSE05x_RSAKeyFormat_CRT);
+	if (status != SM_OK)
+		return kStatus_SSS_Fail;
+
+	return kStatus_SSS_Success;
+}
+
+sss_status_t se050_key_store_set_rsa_key_bin(sss_se05x_key_store_t *store,
+					     sss_se05x_object_t *k_object,
+					     struct se050_rsa_keypair *keypair,
+					     struct se050_rsa_keypub *keypub,
+					     size_t key_bit_len)
+{
+	Se05xPolicy_t policy = { };
+
+	if (!store || !store->session || !k_object)
+		return kStatus_SSS_Fail;
+
+	if (se050_key_exists(k_object->keyId, &store->session->s_ctx))
+		key_bit_len = 0;
+
+	switch (k_object->objectType) {
+	case kSSS_KeyPart_Public:
+		return set_rsa_public(&store->session->s_ctx,
+				      &policy, k_object,
+				      keypub, key_bit_len);
+	case kSSS_KeyPart_Private:
+		if (k_object->cipherType == kSSS_CipherType_RSA)
+			return set_rsa_private_rsa(&store->session->s_ctx,
+						   &policy, k_object,
+						   keypair, key_bit_len);
+
+		if (k_object->cipherType == kSSS_CipherType_RSA_CRT)
+			return set_rsa_private_rsa_crt(&store->session->s_ctx,
+						       &policy, k_object,
+						       keypair, key_bit_len);
+		return kStatus_SSS_Fail;
+	case kSSS_KeyPart_Pair:
+		if (k_object->cipherType == kSSS_CipherType_RSA)
+			return set_rsa_keypair_rsa(&store->session->s_ctx,
+						   &policy, k_object,
+						   keypair, key_bit_len);
+
+		if (k_object->cipherType == kSSS_CipherType_RSA_CRT)
+			return set_rsa_keypair_rsa_crt(&store->session->s_ctx,
+						       &policy, k_object,
+						       keypair, key_bit_len);
+		return kStatus_SSS_Fail;
+	default:
+		return kStatus_SSS_Fail;
+	}
+}
+
+sss_status_t  se050_get_free_memory(pSe05xSession_t ctx, uint16_t *p,
+				    SE05x_MemoryType_t type)
+{
+	if (p && ctx && Se05x_API_GetFreeMemory(ctx, type, p) == SM_OK)
+		return kStatus_SSS_Success;
+
+	return kStatus_SSS_Fail;
+}
+
+sss_status_t se050_scp03_send_rotate_cmd(pSe05xSession_t ctx,
+					 struct s050_scp_rotate_cmd *cmd)
+{
+	uint8_t rsp[64] = { 0 };
+	size_t rsp_len = sizeof(rsp);
+	tlvHeader_t hdr = {
+		.hdr = { [0] = 0x80,
+			 [1] = 0xd8,
+			 [2] = 0,
+			 [3] = PUT_KEYS_KEY_IDENTIFIER,
+		},
+	};
+	smStatus_t st = SM_NOT_OK;
+
+	if (!ctx || !cmd)
+		return kStatus_SSS_Fail;
+
+	hdr.hdr[2] = cmd->cmd[0];
+	st = DoAPDUTxRx_s_Case4(ctx, &hdr, cmd->cmd, cmd->cmd_len,
+				rsp, &rsp_len);
+
+	if ((rsp_len - 1 > sizeof(rsp)) || rsp_len < 2)
+		return kStatus_SSS_Fail;
+
+	st = (rsp[rsp_len - 2] << 8) + rsp[rsp_len - 1];
+	if (st != SM_OK)
+		return kStatus_SSS_Fail;
+
+	if (!memcmp(rsp, cmd->kcv, cmd->kcv_len))
```

Comment:
it is not really needed since all the dimensions have been calculated beforehand for the length of the AES predefined keys, so everthing just matches...but I can add the check if you want.